language-support/hs/bindings/src/DA/Ledger/Stream.hs

```diff
@@ -120,3 +123,12 @@ streamToList stream = do
         Left (Abnormal s) -> fail s
         Left EOS -> return []
         Right x -> fmap (x:) $ streamToList stream
+
+-- Generate a stream in an asyncronous thread
+asyncStreamGen :: (Stream a -> IO ()) -> IO (Stream a)
+asyncStreamGen act = do
+    stream <- newStream
```

Comment:
This is not save in the presence of exceptions. You can get killed between line 131 and line 133 and you will leak the thread. You need to at least mask exceptions around 131-133. Another option would be to turn this into a callback-style function, e.g., `asyncStreamGen :: (Stream a -> IO ()) -> (Stream a -> IO r) -> IO r` which would allow you to use `withAsync` which is much easier than trying to get this right yourself.

Reply:
I would prefer to switch entirely to the continuation style `with..`, but currently my `nim` and `chat` apps are not structured to make use of `withGetTransactions`, but instead use `getTransactions` and explicitly close the stream. I'm not this is easy/possible to change. Which means I think we will need to keep the non-continuation style code for `asyncStreamGen`.

Reply:
Added `mask_` around the entire body of `asyncStreamGen`